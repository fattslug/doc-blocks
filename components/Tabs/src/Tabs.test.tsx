import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Tabs } from ".";

const TabStub = () => (
  <Tabs>
    <Tabs.Title id="one">Title 1</Tabs.Title>
    <Tabs.Content id="one">Content 1</Tabs.Content>
    <Tabs.Title id="two">Title 2</Tabs.Title>
    <Tabs.Content id="two">Content 2</Tabs.Content>
  </Tabs>
);

afterEach(cleanup);

describe("Tabs", () => {
  test("It matches the snapshot", () => {
    const { container } = render(<TabStub />);
    expect(container).toMatchSnapshot();
  });

  test("It renders both tabs", () => {
    const { queryByText } = render(<TabStub />);
    expect(queryByText("Title 1")).toBeInTheDocument();
    expect(queryByText("Title 2")).toBeInTheDocument();
  });

  test("It renders first tab content by default", () => {
    const { queryByText } = render(<TabStub />);
    expect(queryByText("Content 1")).toBeInTheDocument();
    expect(queryByText("Content 2")).not.toBeInTheDocument();
  });

  test("It changes tab content when unselected title is clicked", () => {
    const { queryByText } = render(<TabStub />);
    expect(queryByText("Content 2")).not.toBeInTheDocument();

    userEvent.click(screen.getByText("Title 2"));

    expect(queryByText("Content 2")).toBeInTheDocument();
  });

  test("It changes tab content when unselected title is key pressed", () => {
    const { queryByText } = render(<TabStub />);
    expect(queryByText("Content 2")).not.toBeInTheDocument();
    expect(queryByText("Title 2")).not.toHaveClass("selected");

    userEvent.type(screen.getByText("Title 2"), "{space}");

    expect(queryByText("Content 2")).toBeInTheDocument();
  });
});
