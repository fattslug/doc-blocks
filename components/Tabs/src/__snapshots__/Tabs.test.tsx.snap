// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs It matches the snapshot 1`] = `
<div>
  <div
    class="src___StyledDiv-sc-1xojtv1-0 kfHqKN"
  >
    <div
      class="src___StyledDiv2-sc-1xojtv1-1 keYmsN"
    >
      <div
        class="src___StyledDiv3-sc-1xojtv1-2 bhJGbF title-1-default title-1-selected"
        id="one"
        role="button"
        tabindex="0"
      >
        Title 1
      </div>
      <div
        class="src___StyledDiv3-sc-1xojtv1-2 lhVIJV title-2-default "
        id="two"
        role="button"
        tabindex="0"
      >
        Title 2
      </div>
    </div>
    <div
      class="src___StyledDiv4-sc-1xojtv1-3 bNSTYn"
    >
      Content 1
    </div>
  </div>
</div>
`;
